### PR TITLE
NETSCRIPT: Add .script deprecation notice & migration guide

### PR DESCRIPTION
**Pull request by LJNeon**
I don't feel the need to justify or explain this change as it has already been agreed to unanimously on Discord.

I can't test this PR, both the migration guide and the terminal warning need to be tested before merge.

### CODE REVIEW

**Comment by LJNeon** [comment]
Also I am unable to run prettier, hopefully that can be done via maintainer edits which should be enabled.

File: src/Documentation/doc/migrations/ns2.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Netscript 2 Migration Guide
+
+With the deprecation of Netscript 1 there are required changes to migrate over.
```
Comment (Alpheus) [inline src/Documentation/doc/migrations/ns2.md]: Add a hint to whom this migration doc is for, ie.
"... if you have any .script files." If all your scripts are .js you do not need to follow this guide.

File: src/Documentation/doc/migrations/ns2.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    } else {
+      await ns.hack(target);
+    }
+  }
+}
+```
+
+## Additional problems or edge cases
+
+To get additional help with the migration join the [official Discord server](https://discord.gg/TFc3hKD).
```
Comment (Alpheus) [inline src/Documentation/doc/migrations/ns2.md:78]: Great shoutout for discord!

File: src/Terminal/commands/common/editor.ts [hunk 3]
```diff
@@ -47,10 +49,17 @@ export function commonEditor(
...
+    if (path.endsWith(".script")) hasNs1 = true;
     const file = server.getContentFile(path);
     const content = file ? file.content : isNs2(path) ? newNs2Template : "";
     files.set(path, content);
     if (content === newNs2Template) CursorPositions.saveCursor(path, { row: 3, column: 5 });
   }
+  if (hasNs1) {
+    Terminal.warn(
+      "NOTICE: NS1 (.script) scripts are deprecated and will be removed in a future update." +
+      " Migrate to NS2 (.js) scripts instead."
```
Comment (Alpheus) [inline src/Terminal/commands/common/editor.ts]: Please add a comment to seek the migration guide in the documentation, ie.
" Migrate to NS2 (.js) scripts following the migration guide in the documentation."

File: src/Terminal/commands/runScript.ts [hunk 4]
```diff
@@ -52,6 +52,12 @@ export function runScript(path: ScriptFilePath, commandArgs: (string | number |
   const success = startWorkerScript(runningScript, server);
   if (!success) return Terminal.error(`Failed to start script`);
 
+  if (path.endsWith(".script")) {
```
Comment (Alpheus) [inline src/Terminal/commands/runScript.ts:56]: We'll have to hunt these migration notices down to remove them once deprecation is final. 

Extract to a migration ts file or leave a comment so the search for them can be semi-automated.

Comment (LJNeon) [reply src/Terminal/commands/runScript.ts:56]: They are only in two places, the commonEditor function and the runScript function. Also it's extremely easy to find them by searching "migrate", I just checked.

File: src/Documentation/doc/migrations/ns2.md (hunk 1 above)
Comment (LJNeon) [reply src/Documentation/doc/migrations/ns2.md]: Good idea, done.

File: src/Terminal/commands/runScript.ts (hunk 4 above)
Comment (d0sboots) [reply src/Terminal/commands/runScript.ts:56]: I agree with idea of extraction, for different reasons: we might want to adjust the message, or the criteria (maybe only the first time you edit?) and that should be kept in sync.

Comment (LJNeon) [reply src/Terminal/commands/runScript.ts:56]: I've moved it out into it's own file, that should make modifications trivial.

**Review by Snarling — COMMENTED** [review]
The terms "ns1" and "ns2" are not really accurate and should not be used. .script are a legacy script format that is no longer supported, and .js are javascript files.

The original game documentation pretended that "Netscript" was the language used by game scripts. It's not it's own language, and it's helpful for players if they know the scripts they are writing are literally javascript. `.script` files were an interpreted version of javascript based on es5, whereas `.js` files are actual javascript modules running natively within the player's browser. We intentionally removed references to "ns1" and "ns2" a long time ago.

Netscript is still the name of the game's API, which is passed in as the first argument for player script `main` functions with the default parameter name `ns`. But there is no "ns1" or "ns2", as both script formats can access the same API (although some parts of it don't work in .script files)

**Comment by LJNeon** [comment]
So you want to go by just file extensions instead? Or do you want that specific "legacy script format" and "javascript files" distinction?

**Comment by Alpheus** [comment]
> So you want to go by just file extensions instead? Or do you want that specific "legacy script format" and "javascript files" distinction?

It seems the distinction is in the "execution format" being deprecated, not anything Netscript-related. Netscript is the API and it stays. In that context the message applies to: "deprecating the ability to run '.script' files to access the Netscript API".

**Comment by LJNeon** [comment]
@Snarling is that wording better?

**Comment by Snarling** [comment]
@LJNeon I made some additional changes and added some more content, mind looking this over before I merge it? Sorry for the delay in following up on this.

**Comment by LJNeon** [comment]
Looks good to me!

**Pull request merged**